protocol-units/bridge/contracts/src/NativeBridge.sol

```diff
@@ -120,7 +135,41 @@ contract NativeBridge is AccessControlUpgradeable, PausableUpgradeable, INativeB
         emit BridgeTransferCompleted(bridgeTransferId, initiator, recipient, amount, nonce);
     }
 
+    function setRateLimits(uint256 _outboundRateLimit, uint256 _incomingRateLimit) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        outboundRateLimit = _outboundRateLimit;
+        incomingRateLimit = _incomingRateLimit;
+        emit RateLimitsUpdated(_outboundRateLimit, _incomingRateLimit);
+    }
+
     function togglePause() external onlyRole(DEFAULT_ADMIN_ROLE) {
         paused() ? _pause() : _unpause();
     }
+
+    // _rateLimitOutbound(uint256 amount) internal {
+    //     (uint256 year, uint256 month, uint256 day) = block.timestamp.timestampToDate();
+    //     require(outboundRateLimitBudget[year][month][day] += amount <= outboundRateLimit, OutboundRateLimitExceeded());
+    // }
+
+    // _rateLimitIncoming(uint256 amount) internal {
+    //     (uint256 year, uint256 month, uint256 day) = block.timestamp.timestampToDate();
+    //     require(incomingRateLimitBudget[year][month][day] += amount <= incomingRateLimit, IncomingRateLimitExceeded());
+    // }
+
+    _rateLimitOutbound(uint256 amount) internal {
+        _resetTime();
+        require(outboundRateLimitBudget += amount <= outboundRateLimit, OutboundRateLimitExceeded());
+    }
+
+    _rateLimitIncoming(uint256 amount) internal {
+        _resetTime();
+        require(incomingRateLimitBudget += amount <= outboundRateLimit, IncomingRateLimitExceeded());
+    }
+
+    _resetTime() internal {
```

Comment:
This is too naive a way to do this. You would want a sliding window. 

Consider the case where you have a large volume of transfers immediately before one day since the last reset at $t-1$. You reset at $t$. Then, you have not deducted these from your budget at $t+1$. 

As the granularity of your tracking approaches the duration risk window, the value of the rate limit loses meaning. 

However, actuality, you don't need to reset. The rate limit is de facto provided by the balance of the insurance fund. The notion of the risk period or a daily rate limit is only for our operational perspective which is where the $\text{rateLimit} = \frac{\text{securityFund}}{2 * \text{riskPeriod}}$ comes from. As long as you are computing your rate limit affecting an individual transaction at any point in time according to this, you are safe under the operational framework of MIP-56. 

In other words, the budget will always be whatever we provide. 

What elements of the time boundary you would include here would be for fairness amongst users. So, if you have

Reply:
yes. this has been bugging me before as well and i thought that it introduces potential for unsafe situations. a sliding window would improve the situation

Reply:
Yeah, I agree that if you keep that calculation in, it's safer.

What I would worry about though is now you're using terms that I thought were supposed to account for wanting the L2 to be covered and for availability of insurance funds for future risks periods just to account for safety in one window.

Sliding window is, ofc, just about how you are actually tracking inflight w.r.t to the above. The safest sliding window is naturally one bucketing on each block. This could actually not be that gas expensive to compute.

When you do not subject a transfer to the rate limit, you would do something like...
```
windows[block.timestamp] += transferAmount
```

In an ideal world, when you call a reset, you would just do something like...
```
gasToDeduct = windows[block.timestamp - 1 days]
totalUsed -= gasToDeduct
```

However, you're not likely to have timestamps line up like that. So, you will want to look back over a few timestamps. This is where using a larger window, i.e. slot, and mapping multiple blocks into a slot could help.

Reply:
You also have to account for potentially not having a transfer in each window, and thus a need to gc--going back further.

Reply:
For this reason, it may be better for MVP to use relatively large slots to divide the risk period into $n$ fractions and simply sum over $n$ totals.

Reply:
Basically, I would encourage you to come to something safer via a combination of a heuristics treatment of the terms in MIP-56 and more granular tracking of totals.

Reply:
For example, if you wanted to leave this entirely heuristic, computing  $\text{rateLimit} = \frac{\text{securityFund}}{4 * \text{riskPeriod}}$ would be basically the same as having two windows per risk period.